### ISSUE TITLE
most recent .npy schedule file saved before .pt model weight file

### ISSUE THREAD

**Issue by lcy5058** [Yuanhy1997/SeqDiffuSeq#9]
You have to use the most recent .npy schedule file saved before .pt model weight file.
For this sentence, it means Must the pt weight file be greater than the npy noise schedule file? Can't you use the same time step parameters? What happens when you use the command "bash ./inference_scrpts/iwslt_inf.sh path-to-ckpts/ema_0.9999_280000.pt path-to-save-results path-to-ckpts/alpha_cumprod_step_280000.npy
"

**Comment by Yuanhy1997** [comment]
uhh, you can try with the time step alpha cumprod as the model ckpt, and I think the results won't fluctuate much since the alpha cumprod won't change sharply. My consideration for this setting is to make the denoising function (the neural net) accommodate to the newly updated noise schedule better. 

BTW for the hyperparameter 20000 for updating the schedule, it just selected intuitively. You may try other choices. We dont have much resource for running the parameter searching experiments. Maybe other hyperparameter settings may boost the performance further. XD

**Comment by lcy5058** [comment]
Good thanks. I noticed that the weight files after the training of such diffusion lm include "ema_xx.pt" and model_xx.pt files. I know that one is the encoding weight of word embedding and the other is the training weight, but it seems that only the former is used in the code when reasoning. Is the latter not used? Or is there another reason

**Comment by Yuanhy1997** [comment]
ema pt and model pt all contain the full model params individually. Ema is the moving averaged params during training while Model pt is just the params updated at the time step. EMA is a common trick to boost performance.

**Comment by lcy5058** [comment]
Well, thank you very much. So I can use model.pt in the experiment, but it may not be as effective as ema.pt? And I can only save the ema file and discard the model file

**Comment by Yuanhy1997** [comment]
Of course, while I think it won't hurt to save both : )

**Comment by lcy5058** [comment]
Hello, it was disturbed again. 
What does eval_loss_q1, q2 and q3 recorded in the log file refer to? How is it different from Eval_loss? In your best training results, are you refer to Eval_loss or EVAL_MSE?

**Comment by Yuanhy1997** [comment]
The number after q's refer to the quarters of diffusion steps. The eval loss is the overall loss. The loss refers to the MSE loss per diffusion step.

**Comment by lcy5058** [comment]
Okay, so when the prediction results are the best case, the EVAL_LOSS should reach the minimum instead of Eval_mse

**Comment by Yuanhy1997** [comment]
I think minimum eval loss is not necessarily correlated with the best generation results. In my opinion, this is a pretty similar circumstance where the lowest PPL does not translate to best generation results in AR models. The eval loss includes mse loss and other CE loss and regulations.
